plugins/event_notification/providers/bpm/lib/model/BusinessProcessSignalNotificationTemplate.php

```diff
@@ -19,9 +19,44 @@ public function __construct()
 	 */
 	public function dispatch(kScope $scope)
 	{
-		return $this->dispatchPerCase($scope);
+		$jobData = $this->getJobData($scope);
+		/* @var $jobData kBusinessProcessNotificationDispatchJobData */
+		if(!$jobData->getObject())
+		{
+			return;
+		}
+
+		// check if event template is dispatching from a batch job event, if so, try to get the right case to signal
+		$currentCaseId = null;
+		if ($scope instanceof kEventScope && $scope->getEvent() instanceof IKalturaBatchJobRelatedEvent)
+		{
+			/** @var IKalturaBatchJobRelatedEvent $currentEvent */
+			$currentEvent = $scope->getEvent();
+			$eventBatchJob = $currentEvent->getBatchJob();
+			$eventJobData = $eventBatchJob->getData();
+			// kIntegrationJobData is not one of our dependant plugins
+			if (class_exists('kIntegrationJobData') && $eventJobData instanceof kIntegrationJobData)
+			{
+				$eventJobTriggerData = $eventJobData->getTriggerData();
+				if ($eventJobTriggerData instanceof IBusinessProcessCaseIdRelated)
+					$currentCaseId = $eventJobTriggerData->getCaseId();
+			}
+		}
+
+		$caseIds = $this->getCaseIds($jobData->getObject());
+		$jobId = null;
+		foreach($caseIds as $caseId)
+		{
+			// when we have $currentCaseId, we should only signal to that case
+			if (!is_null($currentCaseId) && $caseId !== $currentCaseId)
+				continue;
+
+			$currentJobData = clone $jobData;
+			$currentJobData->setCaseId($caseId);
+			$this->dispatchJob($scope, $currentJobData);
```

Comment:
We might have a future use case to signal all cases, but I don't think it's very important for now as all our signaling will have a case id.